### DISCUSSION TITLE
Angular项目集成中遇到的问题

### DISCUSSION THREAD

**Discussion by zk922** [dream-num/univer#2033] [Q&A | 常见问题]
首先感谢作者团队提供了如此优秀的开源项目，我遇到的问题是angular项目中再集成后，ng build构建打包时候遇到的。

![f73300947b95e8060de486201fa6a5ff](https://github.com/dream-num/univer/assets/12084127/1edceb55-901d-49df-ac4e-18cb10299793)

经过检查engine-render构建出来的代码后，发现是编译打包后的代码中有遇到require("fs")。虽然会判断是不是浏览器环境，运行时不会执行到require，但通过不了ng build前的静态检查。

![a0320b76a3fdb0ed7de131f11a954384](https://github.com/dream-num/univer/assets/12084127/43d34907-e879-415f-ae89-6701976fc466)

版本是跟随dev分支的最新代码，我们是本地构建发布到本地仓库的。经检查0.1.7release版本并没有这个问题，可能是后续开发中的改动导致的。

请问有什么解决思路么？感谢。

**Comment by jikkai** [comment]
是否有为 tsconfig.json 配置 "skipLibCheck": true？

**Reply by zk922** [reply]
已经设置了。这一步好像是项目构建前的静态检查，angular的构建现在把配置都隐藏了，目前也没找到angular里给fs polyfill的方法，我自己先建个叫fs项目假装导出一个模块骗过去了。

想了想，这个也不算univer的问题，univer应该也可以支持非ui方式的运行，感谢了

**Comment by wzhudev** [comment]
在 Next.js 中可以使用如下配置将对应的 resolve 设置为空：

```js
/** @type {import('next').NextConfig} */
export default {
  webpack: (config, { isServer }) => {
    if (!isServer)
      config.resolve.fallback = { fs: false, path: false }

    return config
  },
}
```

Angular 应该也是类似。

**Reply by wzhudev** [reply]
另外你的项目是否配置了 Angular 的 SSR？

**Reply by zk922** [reply]
感谢回复。
查了半天angular的构建配置，没有在angular.json配置里面找到忽略模块引用的办法...
再底层的构建工具的配置angular隐藏的有点深。
我手动创建了一个叫fs的项目，导出一个空模块，引进了项目里面来，倒是能骗过angular的静态检查。感觉这个是应该去angular cli那边提个issue...

**Comment by jikkai** [comment]
In webpack 4, may need to set it up like this:

```
externals: {
  fs: 'commonjs2 node:fs'
},
```
